src/ast/nodes/Identifier.ts

```diff
@@ -15,18 +15,34 @@ import LocalVariable from '../variables/LocalVariable';
 import Variable from '../variables/Variable';
 import * as NodeType from './NodeType';
 import SpreadElement from './SpreadElement';
-import { ExpressionEntity, LiteralValueOrUnknown } from './shared/Expression';
+import { ExpressionEntity, LiteralValueOrUnknown, UnknownValue } from './shared/Expression';
 import { ExpressionNode, NodeBase } from './shared/Node';
 import { PatternNode } from './shared/Pattern';
 
 export type IdentifierWithVariable = Identifier & { variable: Variable };
 
+const tdzInitTypesToIgnore = {
+	__proto__: null,
+	ArrowFunctionExpression: true,
+	ClassExpression: true,
+	FunctionExpression: true,
+	ObjectExpression: true
+};
+
+const tdzVariableKinds = {
+	__proto__: null,
+	class: true,
+	const: true,
+	let: true,
+	var: true
+};
+
 export default class Identifier extends NodeBase implements PatternNode {
 	name!: string;
 	type!: NodeType.tIdentifier;
-
 	variable: Variable | null = null;
 	protected deoptimized = false;
+	private isTDZAccess = false;
```

Comment:
This is a tri-state property. For the caching to work in:

https://github.com/rollup/rollup/blob/41d5fde334c814d10d913fab1f9ed8ed701dc191/src/ast/nodes/Identifier.ts#L213-L215

it needs to be:
```
private isTDZAccess: boolean | undefined = undefined;
```
which resolves the 11 npm test failures in 41d5fde334c814d10d913fab1f9ed8ed701dc191.

Reply:
Ah, that makes a lot of sense